src/dev/mocha/__tests__/junit_report_generation.js

```diff
@@ -62,7 +62,7 @@ describe('dev/mocha/junit report generation', () => {
 
     mocha.addFile(resolve(PROJECT_DIR, 'test.js'));
     await new Promise((resolve) => mocha.run(resolve));
-    const report = await fcb((cb) => parseString(readFileSync(XML_PATH), cb));
+    const report = await promisify(parseString)(readFileSync(XML_PATH));
```

Comment:
We should just use `readFile` from `fs/promises` and `parseStringPromise` from `xml2js`.

```suggestion
    const report = await parseStringPromise(await readFile(XML_PATH));
```